src/Database/TigerBeetle/Raw/Account.hs

```diff
@@ -9,39 +9,36 @@ import Database.TigerBeetle.Internal.FFI
 import Foreign.Marshal.Alloc
 import Foreign.Ptr
 import Foreign.Storable
+import Data.Vector qualified as V
+import Database.TigerBeetle.Internal.FFI.Account (TBAccount(..))
 
-createAccounts :: [Account] -> IO (Ptr Packet)
+createAccounts :: [Account] -> IO (Ptr TBPacket)
 createAccounts accounts = do
   accountData <- pack accounts
   packetPtr <- newPacket
   poke packetPtr
-    $ Packet
-    { next         = nullPtr
-    , userData     = nullPtr -- TODO: This should be a generated
-                             -- id from client state
-    , operation    = fromIntegral $ fromEnum CreateAccounts
-    , status       = Ok
-    , dataSize     = fromIntegral $ sizeOf accountData
-    , packetData   = castPtr @TbAccount @() accountData
-    , batchAllowed = 0
-    , batchNext    = nullPtr
-    , batchTail    = nullPtr
-    , batchSize    = 0
-    , reserved     = []
+    $ TBPacket
+    { tbPacketUserData         = nullPtr
+    , tbPacketData             = castPtr @TBAccount @() accountData
+    , tbPacketDataSize         = fromIntegral $ sizeOf accountData
+    , tbPacketUserTag          = 0
+    , tbPacketOperation        = fromIntegral $ fromEnum CreateAccounts
+    , tbPacketStatus           = Ok
+    , tbPacketOpaque           = V.empty
```

Comment:
nit: This array is a fixed size, we could use `Data.Array` here instead of `vector` and avoid the (small) memory overhead.